### ISSUE TITLE
feature: searchable bookmarks for threads and posts

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/nimforum#182]

**Comment by dom96** [comment]
Isn't that something provided by your web browser? :)

**Comment by timotheecour** [comment]
no, chrome doesn't allow bookmarking individual posts (see screenshot), unless you open the individual post first (not just the thread!) in a new page and then bookmark that; 

![image](https://user-images.githubusercontent.com/2194784/42713723-4bb533c4-86a5-11e8-9252-4181cb717078.png)

However, the resulting browser bookmark for the individual post is not very useful as it'll just contain the link (eg https://forum.nim-lang.org/t/4041 or https://forum.nim-lang.org/t/4041#25172) named after the thread (eg no information on the author of that individual post):

https://www.dropbox.com/s/pfvzbs03a26sfzd/Screenshot%202018-07-13%2014.05.37.png?dl=0

Worse, it's not searchable.

Here's my use case:

in nimforum search, I want to search this query:
`bookmarked:true "openArray"` # search all posts / threads I bookmarked
`bookmarked:true "openArray"` # search for "openArray" filtered by all posts / threads I bookmarked
`author:bob bookmarked:true "named return value"` # likewise, but filtered by who authored the post

(syntax modelled after github)
